components/style/matching.rs

```diff
@@ -493,6 +528,20 @@ trait PrivateMatchMethods: TElement {
                 .write()
                 .insert(this_opaque, animation_set);
         }
+
+        // If we have modified animation or transitions, we recascade style for this node.
+        if needs_animations_update || might_need_transitions_update {
+            let mut resolver = StyleResolverForElement::new(
+                *self,
+                context,
+                RuleInclusion::All,
+                PseudoElementResolution::IfApplicable,
+            );
+            let new_primary = resolver.resolve_style_with_default_parents();
```

Comment:
I'm a bit confused about this. Why does this need to `resolve_*`, then re-cascade?

Reply:
You're right. Re-cascading here is bogus. I've fixed this in the latest version of the branch.

Reply:
So... I'm still a bit unclear on why this is the right thing to do. This is not just recascading but also re-selector-matching. I'd expect this to do basically just [this](https://searchfox.org/mozilla-central/rev/9ad88f80aeedcd3cd7d7f63be07f577861727054/servo/components/style/traversal.rs#622-635).

And to be clear, I think this is fine as a stepping-stone if needed, but I don't understand why something like that wouldn't work.

Reply:
You're right. I think I was trying this before, but not properly replacing rules. I'll need to do it a bit more carefully because `replace_rules` is written with the animations-only restyle in mind. Thanks for the feedback.